pkg/kubelet/status/status_manager.go

```diff
@@ -154,6 +177,15 @@ func (m *manager) Start() {
 		return
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+		stateImpl, err := state.NewStateCheckpoint(m.stateFileDirectory, podStatusManagerStateFile)
```

Comment:
Checkpoint is the source of truth for ResourcesAllocated (what kubelet agreed to reserve for the Pod's containers). If checkpoint is deleted, then we lose allocation information and may likely end up with the node oversubscribed in the multiple-scheduler case. 

kube-scheduler uses max(Spec.Resources, Status.ResourcesAllocated) however race condition still exists. Without checkpoint info, we cannot settle the race. 

To avoid node-local checkpointing, our [earlier implementation](https://github.com/kubernetes/kubernetes/pull/92127) was based on ResourcesAllocated as part of the PodSpec and that KEP was initially approved but the API folks had second thoughts about the design with the above implementation very close to the finish line. So we went back to the drawing board and made a compromise and agreed to checkpoint on the node. 

[Going back even further in history, our first design that I presented in Kubecon Barcelona in 2019 was based on scheduler approving resize and avoiding the race. However that design was running afoul of API conventions. If you are bored - the [gory details](https://github.com/kubernetes/enhancements/pull/686) :) ]

In all fairness, Tim is also pushing back to the apimachinery folks about the notion that Status can be lost - needs to be regenerable from observations etc. If we can rely on Status persistence, then we can toss this out at some point.

Reply:
I’m fine with the checkpoint. However, if checkpoint loss will result in a
bad result, we should fail loudly, instead of silently create a new
checkpoint. There probably need to be some simple check to figure out
whether it is checkpoint loss or no checkpoint yet.

I'm not saying that we need to resolve this in alpha, but it at least worths a record and a TODO, so that we can solve the problem later, e.g. in Beta.

Reply:
I'm bubbling up errors and logging them now. I have a TODO to investigate to see if we can recover from it.

Reply:
Please track this in this beta.

Reply:
@vinaykul not sure if this was tracked, but we should make sure there is something captured in kep/feature.